viziphant/events.py

```diff
@@ -63,13 +63,18 @@ def add_event(axes, event, key=None, rotation=40):
     """
     axes = np.atleast_1d(axes)
     times = event.times.magnitude
+
+    # Get the limits of the last Axes object in the list
+    x_lim_min, x_lim_max = axes[-1].get_xlim()
+
     for event_idx, time in enumerate(times):
-        if key is None:
-            label = event.labels[event_idx]
-        else:
-            label = event.array_annotations[key][event_idx]
-        for axis in axes:
-            axis.axvline(time, color='black')
-        axes[0].text(time, axes[0].get_ylim()[1], label,
-                     horizontalalignment='left',
-                     verticalalignment='bottom', rotation=rotation)
+        if x_lim_min <= time <= x_lim_max:
```

Comment:
Optional:
(There might be edge cases, where an event is really close to being inside the limits but is discarded.)
- add a section "Notes" in the doctsring to inform users, that events may be discarded if they are outside the limits ?
- add a warning if events are discarded for this reason ?

Those are just ideas, what do you think ?

Reply:
If I understood correctly, the idea is to have a certain tolerance and check if there was an event close by outside the boundary and warn? From my perspective, the idea of the function is to discard events smaller than the lower limit and larger than the highest, as they are shown as those floating labels. So I am not sure about the benefit of such warnings.

Reply:
I agree, warning is not appropriate for this case.

Reply:
- add a section "Notes" in the doctsring to inform users, that events may be discarded if they are outside the limits ?